src/transformers/generation/flax_utils.py

```diff
@@ -334,6 +338,9 @@ def generate(
         # set init values
         prng_key = prng_key if prng_key is not None else jax.random.PRNGKey(0)
 
+        if generation_config.num_return_sequences > generation_config.num_beams:
```

Comment:
Maybe we can put this check in the `elif` part of the loop for beam search, i.e. after the line:
```python
elif not generation_config.do_sample and generation_config.num_beams > 1:
```
Just to separate out the greedy/sample/beam logic!

Reply:
Good Idea. Done!